### PR TITLE
[Perf][Qwen3-Omni]Optimize TTFP using initial_codec_chunk_frames

### PR DESCRIPTION
**Pull request by amy-why-3459**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose
Optimize TTFP using initial_codec_chunk_frames
## Test Plan
```
export SEED_TTS_WER_EVAL=1
export SEED_TTS_EVAL_DEVICE=cuda:1
export VLLM_USE_FLASHINFER_MOE_FP16=0
vllm bench serve \
    --omni \
  --port 28889 \
  --max-concurrency 16 \
  --dataset-name seed-tts \
  --dataset-path /data/why/seed-tts-eval \
  --num-prompts 2000 \
  --no-oversample \
  --seed-tts-wer-save-items \
  --model /data/why/Qwen3-Omni-30B-A3B-Instruct \
  --endpoint /v1/chat/completions \
  --backend openai-chat-omni \
  --percentile-metrics ttft,tpot,itl,e2el,audio_ttfp,audio_rtf \
  --extra_body '{"modalities": ["text", "audio"]}' >seed_tts.log 2>&1 &
```
## Test Result
TTFP 905ms -> TTFP 356ms

```
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--- Running Summary
=========== 38 passed, 3 skipped, 18 warnings in 2265.28s (0:37:45) ============
sys:1: DeprecationWarning: builtin type swigvarlink has no __module__ attribute
```
```
===== Seed-TTS eval (seed-tts-eval protocol) =====
Evaluated (WER, lower is better):        1088      
Mean WER:                                0.4029    
Median WER:                              0.0000    
Request failed:                          0         
No PCM captured:                         0         
ASR / WER failed:                        0         
==================================================
```

concurrency | RTF(main) | RTF(PR4054) | delta
-- | -- | -- | --
1 | 0.13 | 0.12 | -7.69%
4 | 0.24 | 0.21 | -12.50%
8 | 0.28 | 0.27 | -3.57%
16 | 0.34 | 0.34 | 0.00%
32 | 0.56 | 0.50 | -10.71%

concurrency | TTFP(main) | TTFP(PR4054) | delta
-- | -- | -- | --
1 | 527.95 | 243.97 | -53.79%
4 | 901.01 | 397.62 | -55.87%
8 | 1088.75 | 691.3 | -36.51%
16 | 1434.87 | 713.89 | -50.25%
32 | 2388.63 | 1127.80 | -52.78%

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan. Please provide the test scripts & test commands. Please state the reasons if your codes don't require additional test scripts. For test file guidelines, please check the [test style doc](https://docs.vllm.ai/projects/vllm-omni/en/latest/contributing/ci/tests_style/)
- [ ] The test results. Please paste the results comparison before and after, or the e2e results.
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model. **Please run `mkdocs serve` to sync the documentation editions to `./docs`.**
- [ ] (Optional) Release notes update. If your change is user-facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm_omni/model_executor/stage_input_processors/qwen3_omni.py [hunk 1]
```diff
@@ -849,24 +848,33 @@ def talker2code2wav_async_chunk(
...
+        left_context_size = configured_initial_chunk_size
+        end_index = length + configured_initial_chunk_size
+    else:
+        left_context_size = max(0, min(length - context_length, left_context_size_config))
+        end_index = min(length, left_context_size + context_length)
 
-    codes = torch.tensor(transfer_manager.code_prompt_token_ids[request_id][-end_index:]).transpose(0, 1).reshape(-1)
+    codes = (
+        torch.cat(transfer_manager.code_prompt_token_ids[request_id][-end_index:], dim=0).transpose(0, 1).reshape(-1)
+    )
```
Comment (LHXuuu) [inline vllm_omni/model_executor/stage_input_processors/qwen3_omni.py:869]: L853 appends code_predictor_codes as-is, but the L846 else branch leaves it as a non-tensor — the new torch.cat(..., dim=0) will then raise.

Comment (amy-why-3459) [reply vllm_omni/model_executor/stage_input_processors/qwen3_omni.py:869]: > L853 appends code_predictor_codes as-is, but the L846 else branch leaves it as a non-tensor — the new torch.cat(..., dim=0) will then raise.

Fixed

**Comment by amy-why-3459** [comment]
CI's failure is unrelated to this PR.

**Comment by hsliuustc0106** [comment]
lgtm, I think we need to add this insight of how to choose initial_codec_chunk_frames into our blog

**Review by hsliuustc0106 — APPROVED** [review]
